tools/pckgen

```diff
@@ -302,11 +302,25 @@ rpm_spec_prepare() {
     sed -i "s|<CAS_MODULES_DIR>|$CAS_MODULES_DIR|g" "$RPM_SPECS_DIR/$CAS_NAME.spec"
     sed -i "s/<CAS_HOMEPAGE>/${CAS_HOMEPAGE//\//\\/}/g" "$RPM_SPECS_DIR/$CAS_NAME.spec"
     sed -i "s/<PACKAGE_MAINTAINER>/$PACKAGE_MAINTAINER/g" "$RPM_SPECS_DIR/$CAS_NAME.spec"
+
     if [ "$DEBUG" ]; then
-        echo "---   Debug RPMs will be built as well"
-        sed -i "s/%define debug_package %{nil}//g" "$RPM_SPECS_DIR/$CAS_NAME.spec"
+        echo "---   Debug info will be included and debug packages created as well"
+        local rpm_spec_file="$RPM_SPECS_DIR/$CAS_NAME.spec"
+
+        # We need to find the line responsible for compiling process
+        # (to add proper flag to it), and also the line which blocks
+        # debug packages from being built (to force debug package creation).
+        local make_pattern='%make_build$'
+        local debug_package_pattern='%define debug_package'
+
+        if ! grep "$make_pattern" "$rpm_spec_file" &>/dev/null \
+                || ! grep "$debug_package_pattern" "$rpm_spec_file" &>/dev/null; then
+            error "couldn't find proper macros for debug build in file $rpm_spec_file"
+        fi
+        sed -i "/$make_pattern/ s/$/ DEBUG_PACKAGE=1/g" "$rpm_spec_file"
+        sed -i "/$debug_package_pattern/ s/%define debug_package %{nil}//g" "$rpm_spec_file"
         if [[ $(check_os) =~ suse|sles ]]; then
-            sed -i "s/%prep/%debug_package\n\n\n%prep/g" "$RPM_SPECS_DIR/$CAS_NAME.spec"
+            sed -i "s/%prep/%debug_package\n\n\n%prep/g" "$rpm_spec_file"
```

Comment:
This series of `sed` commands greatly obfuscate way in which spec is generated without leaving a hint within the spec itself. I'd rather have template strings within spec file (like `<CAS_NAME>` or `<CAS_VERSION>`) that are populated by `pckgen` depending on its parameters. That way parts of the spec that are generated dynamically will be clearly separated from parts that are static.

Reply:
done